### PR TITLE
PCHR-2153: Update modal window with new design and adding new fields

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
As per the new wireframe of New Document modal,  newly added fields "Valid From, Document Number and Remind Me" fields needs to be integrated into New Document Modal in Task ad Assignment dashboard. And also update the design of the modal.

## Before
Previous design without above mentioned new fields.
1. New Document Modal
screen shot 2017-05-02 at 7 15 43 pm

## After
New Document Modal with new Design and all three new fields integrated:
### A) Add Mode
1. With Detail Tab 
screen shot 2017-05-02 at 8 26 06 pm

2. With Assignment tab
screen shot 2017-05-02 at 7 17 32 pm

### B) Edit Mode
1. Details Tab
screen shot 2017-05-02 at 8 28 18 pm

2. Assignment Tab
screen shot 2017-05-02 at 8 28 28 pm

### Comments:
Note: This PR is for the subtask of PCHR-2145. So, this is pointing to its parent ticket branch
`PCHR-2145-updates-to-documents` for now. When all sub task will be finished `PCHR-2145-updates-to-documents` will be merged into staging as discussed with @mickadoo .

- [x] Tests Pass

### CODE REVIEW

**Comment by mi-kon** [comment]
@bmnepali The first date should be "From" not "form"

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 1]
```diff
@@ -79,6 +93,55 @@ define([
       });
     });
 
+    describe('addAssignee()', function () {
+      beforeEach(function () {
+        initController();
+        addAssignee(sampleAssignee);
+      });
+
+      it('adds contact of id 5 as assignee for document', function () {
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: better wording
`adds contacts to list of assignees`

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 2]
```diff
@@ -79,6 +93,55 @@ define([
...
+        expect($scope.document.assignee_contact_id[0]).toBe(sampleAssignee.id);
+      });
+
+      it('adds contact of id 5 in contact search list', function () {
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: `adds assignee to contact search list`

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 3]
```diff
@@ -79,6 +93,55 @@ define([
...
+        expect($scope.document.assignee_contact_id.length).toEqual(1);
+        $scope.removeAssignee(0);
+        expect($scope.document.assignee_contact_id.length).toEqual(0);
+      });
+    });
+
+    describe('remindMeInfo()', function () {
+      beforeEach(function () {
+        initController();
+
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: empty line?

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 4]
```diff
@@ -79,6 +93,55 @@ define([
...
+        spyOn(CRM,'help');
+
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: empty line?

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 5]
```diff
@@ -79,6 +93,55 @@ define([
...
+        $scope.remindMeInfo();
+      });
+
+      it('makes calls to CRM.help() to display help message', function () {
+        expect(CRM.help).toHaveBeenCalled();
+        expect(CRM.help).toHaveBeenCalledWith('Remind me?', $scope.remindMeMessage, 'error');
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: toHaveBeenCalledWith implies toHaveBeenCalled is called so toHaveBeenCalledWith is enough

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 6]
```diff
@@ -8,6 +8,19 @@ <h4 class="modal-title">New Document</h4>
   </div>
   <div class="modal-body" nv-file-drop uploader="uploader">
     <div class="container">
+      <div class="row">
+        <div class="col-xs-12 col-sm-7"></div>
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: what is the purpose of this div?

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 7]
```diff
@@ -8,6 +8,19 @@ <h4 class="modal-title">New Document</h4>
...
+        <div class="col-xs-12 col-sm-5" ng-show="statusFieldVisible()">
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: why 2 col classes used here?

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 8]
```diff
@@ -37,145 +50,170 @@ <h4 class="modal-title">New Document</h4>
...
+
+    </div>
+    <div class="container">
+      <uib-tabset>
+        <uib-tab heading="Details">
+          <!-- Drop Zone -->
+          <br>
+          <div class="row">
+            <div class="col-xs-12">
+              <div class="well drop-zone" ng-click="dropzoneClick();">
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: is ; required in ng-click?

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 9]
```diff
@@ -37,145 +50,170 @@ <h4 class="modal-title">New Document</h4>
...
-        </div>
-      </div>
-      <!-- Drop Zone -->
-      <div class="row">
-        <div class="col-xs-12">
-          <div class="well drop-zone" ng-click="dropzoneClick();">
-            <i class="fa fa-cloud-upload"></i><br/>
-            <strong>Drop file here</strong><br/> or click to browse
+          <div class="row">
+            <!-- Form Date -->
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: form or from

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 10]
```diff
@@ -37,145 +50,170 @@ <h4 class="modal-title">New Document</h4>
...
-        <div class="col-xs-12">
-          <div text-angular ta-toolbar="[['bold','italics','underline','strikeThrough','ul','ol','undo','redo','clear']]" ng-model="document.details">
+            </div>
           </div>
-        </div>
-      </div>
+          <!-- Assignee & Assignment -->
+          <div class="row">
+            <!-- Assignee -->
+            <div class="col-xs-12 col-sm-7" ng-show="!document.assignee_contact_id[0] && !showAssigneeField">
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: why index 0 for `document.assignee_contact_id[0]` , can the length of array be used?

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 11]
```diff
@@ -37,145 +50,170 @@ <h4 class="modal-title">New Document</h4>
...
+              <ui-select prevent-animations name="assignee" theme="civihr-ui-select" ng-model="document.temp" on-select="addAssignee($item);">
+                <ui-select-match prevent-animations class="ui-select-match" placeholder="Start typing a name or email...">
+                  {{$select.placeholder}}
+                </ui-select-match>
+                <ui-select-choices prevent-animations class="ui-select-choices" repeat="contact.id as contact in contacts.assignee | filter: $select.search" refresh="refreshContacts($select.search, 'assignee')" refresh-delay="0">
+                  <div ng-bind="contact.label"></div>
+                  <small ng-bind="contact.description[0]"></small>
+                </ui-select-choices>
+              </ui-select>
+              <h4 ng-show="document.assignee_contact_id[0]">Assignees:</h4>
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: why index 0?

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html [hunk 12]
```diff
@@ -37,145 +50,170 @@ <h4 class="modal-title">New Document</h4>
...
+              <ul class="list-unstyled">
+                <li ng-repeat="(key,assigneeId) in document.assignee_contact_id" >
+                  <span ng-repeat="contact in cache.contact.arrSearch">
+                    <p ng-if="contact.id == assigneeId">{{ contact.label }}
+                      <a href="javascript:void(0)">&nbsp;<i ng-click="removeAssignee(key)" class="fa fa-times" aria-hidden="true"></i></a>
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: why is ng-click not on <a>?

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js (hunk 1 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: will do

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js (hunk 2 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: will do

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js (hunk 3 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: will fix it.

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 6 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: I was using it to complete the 12 column bootstrap grid, will fix it by adding offset.

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 7 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: Will add offset to fix the grid.

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 8 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: actually no, will fix

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 9 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: Thank you!  for catching, it should be "from".

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 10 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: right! will fix.

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 11 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: will replace by `document.assignee_contact_id.length`

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html (hunk 12 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: I am adding `ng-click` only on the cross button, which is used to remove the assginee form the list of assignees.

Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: sorry, its not on anchor tag .

Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/document.html]: fixed!

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 13]
```diff
@@ -7,20 +7,34 @@ define([
...
 
       data = {};
       files = {};
     }));
 
+    beforeEach(function () {
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: ps document why above calls are required

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js [hunk 14]
```diff
@@ -7,15 +7,22 @@ define([
   'use strict';
 
   describe('ModalDocumentCtrl', function () {
-    var ctrl, modalInstance, $controller, $rootScope, $scope, HR_settings, data, files, initController;
+    var ctrl, modalInstance, $controller, $rootScope, $scope, HR_settings, data, files, initController, $httpBackend, sampleAssignee;
 
     beforeEach(module('civitasks.appDashboard'));
-    beforeEach(inject(function (_$controller_, _$rootScope_) {
+    beforeEach(inject(function (_$controller_, _$rootScope_, _$httpBackend_) {
```
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: please remove httpbackedn if not used

Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: done!

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js (hunk 13 above)
Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js]: removed as we don't need them for the added changes.

File: uk.co.compucorp.civicrm.tasksassignments/js/dist/tasks-assignments.min.js [hunk 15]
```diff
@@ -1,10 +1,997 @@
+define('tasks-assignments/modules/config',[
```
Comment (mickadoo) [inline uk.co.compucorp.civicrm.tasksassignments/js/dist/tasks-assignments.min.js]: It looks like this file wasn't minified.

Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/dist/tasks-assignments.min.js]: Right interesting! I tried running gulp task but still is not minified.

**Pull request merged**